org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullReferenceTest.java

```diff
@@ -18568,4 +18568,130 @@ void m(String packageName, IPS[] packages) {
 			"""
 		});
 }
+// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1461
+public void testGH1461() {
+	if (this.complianceLevel == 0) {
+		buildAllCompliancesTestSuite(testClass());
+		this.complianceLevel = ClassFileConstants.JDK15;
+	}
+	if (this.complianceLevel < ClassFileConstants.JDK15) return;
+	runNegativeTest(
+		new String[] {
+			"X.java",
+			"""
+				import java.util.Objects;
+				public class X {
+					public void test() {
+						String name = null;
+						if (Objects.isNull(name)) {
+							System.out.println("Name is null");
+							return;
+						}
+						System.out.println(name.substring(0, 4));
+					}
+				}
+				"""
+			},
+		"""
+			----------
+			1. WARNING in X.java (at line 9)
+				System.out.println(name.substring(0, 4));
+				^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+			Dead code
+			----------
+			""");
+}
+//https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1461
+public void testGH1461SuppressWarnings() {
+	if (this.complianceLevel == 0) {
+		buildAllCompliancesTestSuite(testClass());
+		this.complianceLevel = ClassFileConstants.JDK15;
+	}
+	if (this.complianceLevel < ClassFileConstants.JDK15) return;
+	runNegativeTest(
```

Comment:
should be `runConformTest()`.

Reply:
> should be runConformTest().

Fixed.

Reply:
> > Thanks. Still Class.isInstance() is not covered, and I believe there's a surprise hidden :)
> 
> Could you, please, help me? Which test?

OK :)

Try something like:
```
Foo foo = new Foo();
if (Bar.class.isInstance(foo)) {
    return;
}
// what do we know about foo here?
```

The point is: `nonNull()` tells about both branches:
* then-branch: reachable if argument is non-null
* else-branch: reachable if argument is null.

But `Class.isInstance()` is not symmetric:
* if argument is null only the else branch is reachable
* if argument is non-null still both branches are reachable

Does this make sense?

Reply:
I have added one test.

Reply:
Thanks, but the test outcome is wrong. I've modified your test like this:
```
public class X {
	public static void main(String... args) {
		Foo foo = new Foo();
		if (Bar.class.isInstance(foo)) {
			return;
		}
		System.out.println("Hello, world!");
	}
}
class Foo{}
class Bar{}
```

Compilation with your changes says:
```
----------
1. WARNING in /tmp/X.java (at line 7)
        System.out.println("Hello, world!");
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dead code
----------
```

But since this is only a warning, I can still execute the programm:
```
$ java X
Hello, world!
```

Ups, the line that was flagged as dead code, **is actually executed**.

Do you now see my point? :smile:

Please also note, that the inverse is good:
```
		Foo foo = null;
		if (!Bar.class.isInstance(foo)) {
			return;
		}
		System.out.println("Hello, world!");
```
Now the println *is dead code*, because `null` is never an instance of `Bar.class`, so we know that `return` is the only possible path.

Reply:
working on it

Reply:
Trying...

Reply:
@stephan-herrmann I have updated the PR. Could you, please, review it?